src/app/content/components/TableOfContents/index.tsx

```diff
@@ -23,28 +24,171 @@ interface SidebarProps {
   page?: Page;
 }
 
+function TabTrapper({
+  mRef,
+  isTocOpen,
+}: {
+  mRef: MutableRefObject<HTMLElement>;
+  isTocOpen: boolean;
+}) {
+  const isPhone = useMatchMobileMediumQuery();
+  const isMobile = useMatchMobileQuery();
+
+  React.useEffect(() => {
+    const refIsSensible = mRef && 'current' in mRef && mRef.current;
```

Comment:
Suggestion: you could move this directly into the if statement instead of creating an one off variable